Adding SNMP exporter for network switch metrics

computate opened the pull request:
This is based on the Bitnami Prometheus SNMP Exporter[1] with basic SNMP
metrics for the new switches.

[1] https://artifacthub.io/packages/helm/prometheus-community/prometheus-snmp-exporter

You can find [snmp_.* related metrics here in NERC Observability Grafana](https://grafana.apps.obs.nerc.mghpcc.org/explore?schemaVersion=1&panes=%7B%22gss%22:%7B%22datasource%22:%22a0fa9d88-8932-41f7-af80-6f678d4fb1e7%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22expr%22:%22snmp_scrape_duration_seconds%22,%22range%22:true,%22instant%22:true,%22datasource%22:%7B%22type%22:%22prometheus%22,%22uid%22:%22a0fa9d88-8932-41f7-af80-6f678d4fb1e7%22%7D,%22editorMode%22:%22code%22,%22legendFormat%22:%22__auto%22%7D%5D,%22range%22:%7B%22from%22:%22now-1h%22,%22to%22:%22now%22%7D%7D%7D&orgId=1)

See also the [SNMP Exporter](https://github.com/prometheus/snmp_exporter)

Generated with this helm command: 
`helm template prometheus-snmp-exporter prometheus-community/prometheus-snmp-exporter -n snmp-exporter --set serviceMonitor.enabled=true --set-json serviceMonitor.params='[{"module":["if_mib"], "name": "vlan-935-moc-swmon", "target": "10.82.3.41"}]' | yq -s '.kind + "-" + .metadata.name'
`

computate committed:
Adding SNMP exporter for network switch metrics

This is based on the Bitnami Prometheus SNMP Exporter[1] with basic SNMP
metrics for the new switches.

[1] https://artifacthub.io/packages/helm/prometheus-community/prometheus-snmp-exporter

larsks reviewed (changes requested):
When deploying snmp-exporter, you need some way of injecting custom configuration, which you usually generate using [snmp-generator](https://github.com/prometheus/snmp_exporter/blob/main/generator/README.md). Those configs are often large, which means they can't be embedded in a configmap. One option is to fetch the generated config from a known url in an `initContainer`.

larsks committed:
Update deployment to use ocp-on-nerc/nerc-snmp-config-generator image

Update the deployment to (a) use the container image we have created for
compiling generator.yml into snmp.yml and (b) modify the initContainer to
read generator.yml from a ConfigMap.

larsks committed:
Add overlay for nerc-ocp-obs

larsks committed:
Add if_mib and ip_mib from snmp_exporter example

computate committed:
Fixed ServiceMonitor modules and metrics

Adding the metrics for the following SNMP modules: if_mib, ip_mib,
cisco_device, cisco_fc_fe, dell, dell_network

larsks committed:
Move allow list to a file

This allows us to validate the YAML in the allowlist, since it's no longer
embedded as a string value.

computate committed:
Add snmp, tcp, udp, and dell switch metrics

Also adding a prefix of network_switch_ to all switch metrics.

computate committed:
Add vendor label and sysName metric for SNMP switches

We will add a hard-coded vendor label to the switch metrics, and a
sysName metric with the host name of the switch.
